rc/filetype/diff.kak

```diff
@@ -11,3 +11,45 @@ hook -group diff-highlight global WinSetOption filetype=diff %{
     add-highlighter window/diff ref diff
     hook -once -always window WinSetOption filetype=.* %{ remove-highlighter window/diff }
 }
+
+define-command \
+    -docstring %{diff-select-file: Select patch file under the cursor} \
+    -params 0 \
+    diff-select-file %{
+    evaluate-commands -itersel -save-regs 'ose/' %{
+        try %{
+            execute-keys '"oZgl<a-?>^diff <ret>;"sZ' 'Ge"eZ'
+            try %{ execute-keys '"sz?\n(?=diff )<ret>"e<a-Z><lt>' }
+            execute-keys '"ez'
+        } catch %{
+            execute-keys '"oz'
+            fail 'Not in a diff file'
+        }
+    }
+}
+
+define-command \
+    -docstring %{diff-select-hunk: Select patch hunk under the cursor} \
+    -params 0 \
+    diff-select-hunk %{
+    evaluate-commands -itersel -save-regs 'ose/' %{
+        try %{
+            execute-keys '"oZgl<a-?>^@@ <ret>;"sZ' 'Ge"eZ'
+            try %{ execute-keys '"sz?\n(?=diff )<ret>"e<a-Z><lt>' }
+            try %{ execute-keys '"sz?\n(?=@@ )<ret>"e<a-Z><lt>' }
+            execute-keys '"ez'
+        } catch %{
+            execute-keys '"oz'
+            fail 'Not in a diff hunk'
+        }
+    }
+}
+
+hook -group diff-text-objects global WinSetOption filetype=diff %{
+    map window -docstring 'diff file' object f '<a-;> diff-select-file<ret>'
```

Comment:
I don't think we should declare mappings in the users' back, providing a command-based API seems good enough.

The docstrings could be better too.

Reply:
I removed the mappings and slightly tweaked the docstrings to use the same language as the object menu (though I'm not sure it's better or what better would be).